geniza/common/middleware.py

```diff
@@ -29,3 +30,20 @@ def __call__(self, request: HttpRequest):
         # otherwise, continue on with middleware chain
         response = self.get_response(request)
         return response
+
+
+class LocaleMiddleware(DjangoLocaleMiddleware):
+    ### customize locale middleware to exempt some urls from redirects
+    # adapted from https://code.djangoproject.com/ticket/17734
+
+    redirect_exempt_paths = ["admin", "annotations", "accounts"]
+
+    def process_response(self, request, response):
```

Comment:
Interesting that overriding `process_response` like this doesn't seem to be in the Django docs! Seems like it's something that could come up a lot.

Reply:
Yeah, I was surprised too. Glad I found the discussion & a solution that was easy to adapt, at least.